pilot/cmd/pilot-agent/main.go

```diff
@@ -325,7 +325,7 @@ var (
 
 			envoyProxy := envoy.NewProxy(proxyConfig, role.ServiceNode(), proxyLogLevel, pilotSAN, role.IPAddresses)
 			agent := proxy.NewAgent(envoyProxy, proxy.DefaultRetry, pilot.TerminationDrainDuration())
-			watcher := envoy.NewWatcher(certs, agent.ConfigCh())
+			watcher := envoy.NewWatcher(proxyConfig.TlsCertsToWatch, agent.ConfigCh())
```

Comment:
I can not see how these files are used other than watch them ans restart envoy.

Reply:
true. dont you have to update the mTLS plugin in pilot to generate envoy TLS contexts to use the right paths?

Reply:
@hzxuzhonghu: correct. The actual certs that envoy uses come from other configs:
1. xDS (pilot-discovery) mTLS certs are defined in static config.
2. All cert info for proxied requests is passed down from xDS config.

Moving forward my understanding is that Envoy's SDS will handle more of #2.

Reply:
my point is XDS still uses the old path /etc/certs.. the code change here only tells pilot agent to watch the new path. but Pilot (discovery) is still using the old one

Reply:
@rshriram We'll open a separate issue/PR for any pilot-discovery changes needed to customize these cert paths if that's ok.

Reply:
No, I think we need fix it here.